Lib/pdb.py

```diff
@@ -444,9 +444,16 @@ def default(self, line):
         if line[:1] == '!': line = line[1:].strip()
         locals = self.curframe_locals
         globals = self.curframe.f_globals
+        readline = None
         try:
             if (code := codeop.compile_command(line + '\n', '<stdin>', 'single')) is None:
                 # Multi-line mode
+                try:
+                    import readline
+                    # Disable tab-completion temporarily
+                    readline.parse_and_bind('tab: self-insert')
```

Comment:
It seems like the `Cmd` base class is a bit more careful about moves like this. For example, maybe we should only do this `if self.use_rawinput and self.completekey == "tab"`? That way we don't change the global `readline` state in an unexpected way during debugging.

What do you think?

Reply:
Yeah we should protect this with ``use_rawinput`` and ``completekey``, however, I'm not sure if checking the `get_completer()` is necessary. The completer should always be `self.complete`. In any case, we should not trigger the completer function in multiline mode - tab should always input tab.

As for importing readline globally, `cmd` does it twice in the code - I think it's because `readline` is only available on some systems. This is more explicit than having a `readline is None` check in the function.

I can move the readline reverting part into the mutiline if statement - so we can avoid changing the readline states in `exec()` - even though it's hard to imagine having useful code there to use `readline`.